### PR TITLE
solutions of set1 problem 2-8

### PR DESCRIPTION
**Pull request by rnehra01**

### CODE REVIEW

**Comment by jaybosamiya** [comment]
I believe that we will be adding value if we wrote a bit about possible issues for some of the challenges, along with the solutions, rather than just placing the code there.

As for the solutions, the first few (28 at the moment) are here: https://github.com/jaybosamiya/cryptopals-solutions

Edit: Updated repo at https://github.com/jaybosamiya/matasano-crypto/

File: cryptopals/set1/4/4.txt [hunk 1]
```diff
@@ -0,0 +1,327 @@
+0e3647e8592d35514a081243582536ed3de6734059001e3f535ce6271032
```
Comment (r-jenish) [inline cryptopals/set1/4/4.txt:1]: Any particular use of this file? If someone wants to run the solution.py (ch4.py), he can download by himself. This feels redundant. Same goes for 7.txt and 8.txt in their respective directory.

Comment (rnehra01) [reply cryptopals/set1/4/4.txt:1]: Okay, I'll remove them along with the additions suggested by Jay.

Comment (jaybosamiya) [reply cryptopals/set1/4/4.txt:1]: I'd actually say, keep it. The reason is that then you have a consistent thing, and you don't need to depend on an external file. Additionally, it is a tiny file compared to how much GH allows us to keep, so I'd say keep it :)

**Pull request closed without merging**